lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -121,6 +121,27 @@ class RecomposeSelectFill_ : public OpRewritePattern<AtenFill_TensorOp> {
     return success();
   }
 };
+
+class RecomposeScalarTensor : public OpRewritePattern<AtenScalarTensorOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenScalarTensorOp op,
+                                PatternRewriter &rewriter) const override {
+    // rewrite aten.scalar_tensor to
+    // prim.NumToTensor.Scalar + aten.to.dtype_layout
```

Comment:
This is a decomposition, so it should be in `DecomposeComplexOps`. The pattern in this file is for matching on a *series* of ops and (usually) collapsing it to a smaller series of ops that allow further optimizations to take place in torch-mlir.